src/core/dirlistjob.cpp

```diff
@@ -62,6 +63,12 @@ void DirListJob::exec() {
             false
     };
     if(enu) {
+        bool bindCutFiles = false;
+        if(cutFilesHashSet_
+                && cutFilesHashSet_->size() < 450 /* give up if more than 450 cut files */) {
```

Comment:
> I don't see any extra CPU usage in the latter case

I searched for what's the best implementation for this particular case and `std::set` was a winner, probably it's very capable with even much bigger numbers - but not tested. I will remove the condition.

> what about adding !cutFilesHashSet_->empty()? Is it needed?

It's tested already in `folder.cpp:693` and `folder.cpp:258` with `Folder::hasCutFiles()`